tailnet/coordinator.go

```diff
@@ -658,6 +713,31 @@ func (c *core) handleRequest(p *peer, req *proto.CoordinateRequest) error {
 	if req.Disconnect != nil {
 		c.removePeerLocked(p.id, proto.CoordinateResponse_PeerUpdate_DISCONNECTED, "graceful disconnect")
 	}
+	if ack := req.TunnelAck; ack != nil {
+		err := c.handleAckLocked(pr, ack)
+		if err != nil {
+			return xerrors.Errorf("handle ack: %w", err)
+		}
+	}
+	return nil
+}
+
+func (c *core) handleAckLocked(src *peer, ack *proto.CoordinateRequest_Ack) error {
+	dstID, err := uuid.FromBytes(ack.Id)
+	if err != nil {
+		// this shouldn't happen unless there is a client error.  Close the connection so the client
+		// doesn't just happily continue thinking everything is fine.
+		return xerrors.Errorf("unable to convert bytes to UUID: %w", err)
+	}
+
+	dst, ok := c.peers[dstID]
```

Comment:
Will get to this tmrw

Reply:
Added some auth preventing clients from sending them. I thought about it a little more, do we really care about agents being able to send arbitrary `ready_for_handshake`s? If the client receives one from an unknown node, it just drops it. I guess this could technically change in the future, but I don't really see any side effects other than adding an entry to a map.

Reply:
I can't think of any attacks right now either, but I think it's a foundational principle that the control plane should not be delivering unauthorized messages.  The latter is secure by design and provides value even as things change, the former does not.

The coordinator knows all the tunnels, so it should check them before delivering a READY_FOR_HANDSHAKE.